app/Http/Controllers/Api/UserController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Resources\User as UserResource;
+use App\Models\User;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Support\Facades\Gate;
+
+class UserController extends Controller
+{
+    /**
+     * Class constructor.
+     *
+     * @throws AuthorizationException Handled by App\Exceptions\Handler.php.
+     */
+    public function __construct()
+    {
+        // Only accept logged in Users.
+        $this->middleware(function ($request, $next) {
+            if ($request->user() === null) {
+                throw new AuthorizationException();
+            }
+            return $next($request);
+        });
+    }
+
+    /**
+     * Return all users as an array
+     *
+     * @throws AuthorizationException Handled by App\Exceptions\Handler.php.
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        $users = User::with(['applicant', 'manager', 'hr_advisor'])->get();
```

Comment:
Do we need the profile data to come back with the user? If we're not using it, I recommend not loading it. Keeping the profiles separate keeps the permissions simpler, conceptually.

Reply:
I am going to have to include them in the query here for the sake of checking Department permissions in the Gate, but this can be cleaned up once #2846 is finished. I can exclude it from the response thanks to the Laravel API Resource classes.